persistence/postgres-persistence/src/main/java/com/netflix/conductor/postgres/config/outbox/OutboxExecutionDAO.java

```diff
@@ -0,0 +1,363 @@
+    }
+
+    @Override
+    @Deprecated
+    public boolean exceedsInProgressLimit(TaskModel task) {
+        return delegate.exceedsInProgressLimit(task);
+    }
+
+    @Override
+    public boolean removeTask(String taskId) {
+        return delegate.removeTask(taskId);
+    }
+
+    @Override
+    public TaskModel getTask(String taskId) {
+        return delegate.getTask(taskId);
+    }
+
+    @Override
+    public List<TaskModel> getTasks(List<String> taskIds) {
+        return delegate.getTasks(taskIds);
+    }
+
+    @Override
+    public List<TaskModel> getPendingTasksForTaskType(String taskType) {
+        return delegate.getPendingTasksForTaskType(taskType);
+    }
+
+    @Override
+    public List<TaskModel> getTasksForWorkflow(String workflowId) {
+        return delegate.getTasksForWorkflow(workflowId);
+    }
+
+    @Override
+    public String createWorkflow(WorkflowModel workflow) {
+        return delegate.createWorkflow(workflow);
+    }
+
+    @Override
+    public String updateWorkflow(WorkflowModel workflow) {
+        String workflowId = delegate.updateWorkflow(workflow);
+        updateWorkflowOutboxTable(workflow);
+        return workflowId;
+    }
+
+    private void updateWorkflowOutboxTable(WorkflowModel workflow) {
+        Preconditions.checkNotNull(workflow, "workflow object cannot be null");
+        boolean terminal = workflow.getStatus().isTerminal();
+
+        try {
+            withTransaction(
+                    tx -> {
+                        if (terminal) {
+                            writeWorkflowToOutbox(tx, workflow);
+                        }
+                    });
+        } catch (NonTransientException e) {
+            LOGGER.error("Writing workflow with id: {} to outbox table failed", workflow.getWorkflowId(), e);
+        }
+    }
+
+    private void updateTasksOutboxTable(TaskModel task) {
+        boolean terminal = task.getStatus().isTerminal();
+
+        try {
+            withTransaction(
+                    connection -> {
+                        if (terminal) {
+                            validateTask(task);
+                            writeTaskToOutbox(connection, task);
+                        }
+                    });
+        } catch (NonTransientException e) {
+            LOGGER.error("Writing task with id: {} to outbox table failed", task.getTaskId(), e);
+        }
+    }
+
+    /**
+     * Writes task data into outbox.
+     * Data are retrieved by debezium connect and used
+     * for generating kafka messages.
+     * Data are afterward deleted from db.
+     * <p>
+     * Columns in the outbox table:
+     * ID - task id
+     * aggregateType - part of the topic name
+     * aggregateId - partition id - workflow id set for workflow status update
+     * payload - json containing workflow id, task id and status of the task
+     * evenType - task/workflow update state differentiator
+     *
+     * @param connection Hicari connection
```

Comment:
It's Hikary I think but this doesn't have to be Hikary, can be any DB connection as connection is a generic Java interface